### PR TITLE
Update Docker info in CONTRIBUTING.md Section 2.7.b

### LINKED ISSUE #7576: Update CONTRIBUTING.md Section 2.7.b with information about Docker
### Overview
We need to update Section 2.7.b with information about Docker, including instructions for keeping the local Docker image up to date by pulling updates from Dockerhub as needed.  

### Action Items
- [x] Open CONTRIBUTING.md in your IDE
- [x] See draft instructions at https://github.com/hackforla/website/issues/3488#issuecomment-2370625800
- [x] Edit CONTRIBUTING.md to incorporate content from the draft, making changes as required so that the format and style are consistent with CONTRIBUTING.md,  and making any changes required to improve clarity.  
- [x] Changes to CONTRIBUTING.md cannot be tested locally, rather they must be tested after pushing the issue branch to your fork of the repository.  Push your issue branch in the usual manner, but before creating the Pull Request, check your updates using this test URL.  Also store the test URL for use in a later step:
```
https://github.com/[REPLACE WITH GITHUB HANDLE]/website/blob/[REPLACE WITH NAME OF ISSUE BRANCH]/CONTRIBUTING.md
```
   (for example: https://github.com/bonniewolfe/website/blob/issue-branch-1234/CONTRIBUTING.md)
- [x] Create a pull request with your changes.  In the Pull Request, after the "Why did you make the changes" section, add this line to help reviewers, replacing the text in brackets (and the brackets) with the test URL from the previous Action Item.
```
For Reviewers: Do not review changes locally, rather, review changes at [REPLACE WITH TEST URL]
```

### Resources/Instructions
Draft changes https://github.com/hackforla/website/issues/3488#issuecomment-2370625800
[CONTRIBUTING.md](https://github.com/hackforla/website/blob/gh-pages/CONTRIBUTING.md)
[Getting started with writing and formatting on GitHub](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github)

### PR DESCRIPTION
**Pull request by jenjenkayi**
Fixes #7576

### What changes did you make?
  
  - Update Docker information in CONTRIBUTING.md Section 2.7.b per draft instructions at #3488

### Why did you make the changes (we will use this info to test)?
  
  - To add instructions for keeping the local Docker image up to date by pulling updates from Dockerhub

For Reviewers: Do not review changes locally, rather, review changes at https://github.com/jenjenkayi/website/blob/update-contributing.md-7576/CONTRIBUTING.md.

### Screenshots of Proposed Changes To The Website (if any, please do not include screenshots of code changes)
 - No visual changes to the website

### CODE REVIEW

**Review by codyyjxn — COMMENTED** [review]
@jenjenkayi  Thank you for taking on this issue. 
I'm not sure why I am not able to see the changes through the link that you provide in the pr. 
It seems like you made the correct changes. Are you able to see the changes on your end from the url that you provided ?

**Review by pluto-bell — CHANGES_REQUESTED** [review]
Great work, everything looks solid 👍 and the link you provided did work for me.

The one thing I noticed is that the hyperlinks at the top of the `CONTRIBUTING.md` page no longer work for sections `2.7b` and `i` due to the title name changes. 

My suggestion is to:

- [ ]  edit the hyperlinks at the top of the `CONTRIBUTING.md` file so that they reference the new title names for `2.7b` and its subsection `i` so they can continue to be easily accessible via the table of contents hyperlinks. 

- [ ] after completing this change double-check your work to ensure that clicking each of those hyperlinks redirect you to sections `2.7b` and `i`, respectively. 

If you have any questions don't hesitate to reach out.

**Comment by pluto-bell** [comment]
> Great work, everything looks solid 👍 and the link you provided did work for me.
> 
> The one thing I noticed is that the hyperlinks at the top of the `CONTRIBUTING.md` page no longer work for sections `2.7b` and `i` due to the title name changes.
> 
> My suggestion is to:
> 
> * [x]   edit the hyperlinks at the top of the `CONTRIBUTING.md` file so that they reference the new title names for `2.7b` and its subsection `i` so they can continue to be easily accessible via the table of contents hyperlinks.
> * [x]  after completing this change double-check your work to ensure that clicking each of those hyperlinks redirect you to sections `2.7b` and `i`, respectively.
> 
> If you have any questions don't hesitate to reach out.


Thanks for making the adjustments, @jenjenkayi! 

I'll let someone who has more authority than me take a look at these notes and make the final decision on subsection `i`:

-  It looks like `2.7.b` subsection `i` isn't getting properly connected still due to the link inside the full string: 
    - `new title:` i. To test your issue branch locally, run the command "docker-compose up" from a terminal window. The website will then be accessible at http://localhost:4000/ . A few notes regarding Docker:
- My thought is to make subsection `i` more like an actual title so we can still easily hyperlink it
   - `previous title:` What if you cannot see your changes locally within Docker?
- Since I don't have jurisdiction, just putting in my 2 cents (:

**Comment by jenjenkayi** [comment]
@pluto-bell Thank you for reviewing! I have checked both links, they are working now.

**Comment by jenjenkayi** [comment]
@codyyjxn Thank you for reviewing! Yes, the changes are up to date from the url provided.

**Comment by santiseccovidal** [comment]
Review ETA: 2 PM 10/26/24
Availability: Friday-Saturday 11 AM-3 PM

**Comment by santiseccovidal** [comment]
-Change "the Docker Desktop application..." with "The Docker Desktop application..." in the first bullet point.

-I would personally keep the last two bullet points from the previous version and add them at the end.

- If the above steps did not resolve your issue, run through the first three steps again, but try resetting your browser's cache before restarting docker (you can also try running http://localhost:4000 in another browser).

- If you still do not see your changes after trying these steps, please feel free to reach out to the team in the #hfla-site Slack channel, or bring up your issue in a dev meeting.

-In the new Docker version when clicking the three dots, I don't see the option "View details" instead I see the option "View packages and CVEs" I would add that just in case.
I don't know if I would add the screenshot also of the new version or just the text.
Screenshot 2024-10-26 161622

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -546,18 +546,36 @@ We urge developers to be cautious using `git add`. In general it is not advisabl
 
 When you've finished working on your issue, follow the steps below to prepare your changes to push to your repository.
 
-##### **i. What if you cannot see your changes locally within Docker?**
+##### **i. To test your issue branch locally, run the command "docker-compose up" from a terminal window. The website will then be accessible at http://localhost:4000 . A few notes regarding Docker:**
 
-If you do not see the changes you applied when you run `docker-compose up`, **do the following**:
+- the Docker Desktop application must be running, otherwise `docker-compose up` will not work
```
Comment (santiseccovidal) [inline CONTRIBUTING.md]: Here I meant to capitalize "the" with "The"

**Comment by jenjenkayi** [comment]
@santisecco Updated per your comments. Please review. Thank you!

**Review by santiseccovidal — DISMISSED** [review]
Hi @jenjenkayi thank you for the updates and adding the screenshot for the new version of docker. It looks great.
Thanks for working on this.

**Comment by jenjenkayi** [comment]
@pluto-bell I have updated the changes. Please review it again. Thank you

**Comment by codyyjxn** [comment]
@jenjenkayi  Thank you for making the changes. Please request a review so i can approve this.

**Review by codyyjxn — DISMISSED** [review]
@jenjenkayi  Thank you for taking this issue on. Keep it up !

**Review by t-will-gillis — CHANGES_REQUESTED** [review]
Hi @jenjenkayi Thank you for working on this! Everything is looking great and it appears that you have satisfied the original issue. There are a couple of edits, however, that I believe we will want to make:
- As @pluto-bell had mentioned earlier, I agree that the header for 2.7.i  should be a title, and currently this reads as a paragraph. I think that the best solution is to change the text on line 55 to something like (and please check that this is formatted correctly):  
  ```  
  - [**i. A few notes regarding Docker:**](#i-a-few-notes-regarding-docker)  
  ```  
  and line 249 to (please remove the space between the link and the period):  
  ```  
  ##### **i. A few notes regarding Docker:**   

  - To test your issue branch locally, run the command "docker-compose up" from a terminal window. The website will then be accessible at http://localhost:4000. 
  ```  
- Also, on line 552 there is a misspelling "...which enables Docker to built the website locally..." Here "built" should be "build"

Again, you followed the issue correctly but would you please make these edits?
Thanks!

**Review by t-will-gillis — APPROVED** [review]
Hey @jenjenkayi This looks great! Thanks for making the edits.

**Pull request merged**